src/Identity/Extensions.Core/src/UserManagerMetrics.cs

```diff
@@ -0,0 +1,273 @@
+            AddExceptionTags(ref tags, exception);
+
+            _createCounter.Add(1, tags);
+        }
+    }
+
+    private static void AddExceptionTags(ref TagList tags, Exception? exception)
+    {
+        if (exception != null)
+        {
+            tags.Add("error.type", exception.GetType().FullName!);
+        }
+    }
+
+    internal void UpdateUser(string userType, IdentityResult? result, UserUpdateType updateType, Exception? exception = null)
+    {
+        if (_updateCounter.Enabled)
+        {
+            var tags = new TagList
+            {
+                { "aspnetcore.identity.user_type", userType },
+                { "aspnetcore.identity.user.update_type", GetUpdateType(updateType) },
+            };
+            AddIdentityResultTags(ref tags, result);
+            AddExceptionTags(ref tags, exception);
+
+            _updateCounter.Add(1, tags);
+        }
+    }
+
+    internal void DeleteUser(string userType, IdentityResult? result, Exception? exception = null)
+    {
+        if (_deleteCounter.Enabled)
+        {
+            var tags = new TagList
+            {
+                { "aspnetcore.identity.user_type", userType }
+            };
+            AddIdentityResultTags(ref tags, result);
+            AddExceptionTags(ref tags, exception);
+
+            _deleteCounter.Add(1, tags);
+        }
+    }
+
+    internal void CheckPassword(string userType, bool? userMissing, PasswordVerificationResult? result, Exception? exception = null)
+    {
+        if (_checkPasswordCounter.Enabled)
+        {
+            var tags = new TagList
+            {
+                { "aspnetcore.identity.user_type", userType },
+            };
+            if (userMissing != null || result != null)
+            {
+                tags.Add("aspnetcore.identity.user.password_result", GetPasswordResult(result, passwordMissing: null, userMissing));
+            }
+            AddExceptionTags(ref tags, exception);
+
+            _checkPasswordCounter.Add(1, tags);
+        }
+    }
+
+    internal void VerifyToken(string userType, bool? result, string purpose, Exception? exception = null)
+    {
+        var tags = new TagList
+        {
+            { "aspnetcore.identity.user_type", userType },
+            { "aspnetcore.identity.token_purpose", GetTokenPurpose(purpose) },
+        };
+        if (result != null)
+        {
+            tags.Add("aspnetcore.identity.token_verified", result == true ? "success" : "failure");
+        }
+        AddExceptionTags(ref tags, exception);
+
+        _verifyTokenCounter.Add(1, tags);
+    }
+
+    internal void GenerateToken(string userType, string purpose, Exception? exception = null)
+    {
+        var tags = new TagList
+        {
+            { "aspnetcore.identity.user_type", userType },
+            { "aspnetcore.identity.token_purpose", GetTokenPurpose(purpose) },
+        };
+        AddExceptionTags(ref tags, exception);
+
+        _generateTokenCounter.Add(1, tags);
+    }
```

Comment:
Do these methods need to check that their counters are enabled, as is done elsewhere?